src/catalog/tests/open.rs

```diff
@@ -495,79 +495,6 @@ async fn test_open(state_builder: TestCatalogStateBuilder) {
     }
 }
 
-#[mz_ore::test(tokio::test)]
-#[cfg_attr(miri, ignore)] //  unsupported operation: can't call foreign function `TLS_client_method` on OS `linux`
-async fn test_persist_unopened_epoch_fencing() {
-    let persist_client = PersistClient::new_for_tests().await;
-    let state_builder = TestCatalogStateBuilder::new(persist_client);
-    test_unopened_epoch_fencing(state_builder).await;
-}
-
-async fn test_unopened_epoch_fencing(state_builder: TestCatalogStateBuilder) {
-    // Initialize catalog.
-    let zdt_deployment_max_wait = Duration::from_millis(666);
-    let state_builder = state_builder.with_default_deploy_generation();
-    {
-        let mut state = state_builder
-            .clone()
-            .unwrap_build()
-            .await
-            // Use `NOW_ZERO` for consistent timestamps in the snapshots.
-            .open(NOW_ZERO(), &test_bootstrap_args())
-            .await
-            .unwrap();
-        // drain catalog updates.
-        let _ = state.sync_to_current_updates().await.unwrap();
-        let mut txn = state.transaction().await.unwrap();
-        txn.set_0dt_deployment_max_wait(zdt_deployment_max_wait)
-            .unwrap();
-        txn.commit().await.unwrap();
-    }
-    let mut openable_state = state_builder.clone().unwrap_build().await;
-
-    // Read config collection with unopened catalog.
-    assert_eq!(
-        zdt_deployment_max_wait,
-        openable_state
-            .get_0dt_deployment_max_wait()
-            .await
-            .unwrap()
-            .unwrap()
-    );
-
-    // Open catalog, which will bump the epoch.
-    let _state = state_builder
-        .clone()
-        .unwrap_build()
-        .await
-        // Use `NOW_ZERO` for consistent timestamps in the snapshots.
-        .open(NOW_ZERO(), &test_bootstrap_args())
-        .await
-        .unwrap();
-
-    // Unopened catalog should be fenced now with an epoch fence.
-    let err = openable_state
-        .get_deployment_generation()
-        .await
-        .unwrap_err();
-    assert!(
-        matches!(
-            err,
-            CatalogError::Durable(DurableCatalogError::Fence(FenceError::Epoch { .. }))
-        ),
-        "unexpected err: {err:?}"
-    );
-
-    let err = openable_state.is_initialized().await.unwrap_err();
-    assert!(
-        matches!(
-            err,
-            CatalogError::Durable(DurableCatalogError::Fence(FenceError::Epoch { .. }))
-        ),
-        "unexpected err: {err:?}"
-    );
-}
```

Comment:
One of the consequences of this change is that an unopened catalog cannot be fenced via the epoch until it's written (and fenced other catalogs itself) at least once.

Reply:
Makes sense!